src/Symfony/Component/DependencyInjection/Extension/Extension.php

```diff
@@ -82,11 +82,23 @@ public function getConfiguration(array $config, ContainerBuilder $container)
         $class = get_class($this);
         $class = substr_replace($class, '\Configuration', strrpos($class, '\\'));
         $class = $container->getReflectionClass($class);
-        $constructor = $class ? $class->getConstructor() : null;
 
-        if ($class && (!$constructor || !$constructor->getNumberOfRequiredParameters())) {
+        if (!$class) {
+            return null;
+        }
+
+        if (!$class->implementsInterface(ConfigurationInterface::class)) {
+            @trigger_error(sprintf('Not implementing "%s" in the extension configuration class "%s" is deprecated since Symfony 4.1.', ConfigurationInterface::class, $class->getName()), E_USER_DEPRECATED);
```

Comment:
Should probably mention that this will be removed in 5.0

Reply:
IIRC, we stopped mentioning this in all our deprecation warnings to reduce the noise, as removal always happen in the next major anyway.